.github/workflows/android.yml

```diff
@@ -0,0 +1,70 @@
+name: Android
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build-android:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        abi: [arm64-v8a, armeabi-v7a, x86_64]
+        api: [21]
+    env:
+      NDK_CHANNEL: r27d
+      ANDROID_ABI: ${{ matrix.abi }}
+      ANDROID_API: ${{ matrix.api }}
+      # Keep build output out of the repo tree
+      NDK_OUT_DIR: ${{ github.workspace }}/out/${{ matrix.abi }}
+      NDK_LIBS_DIR: ${{ github.workspace }}/libs/${{ matrix.abi }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Cache Android NDK
+        id: cache-ndk
+        uses: actions/cache@v4
+        with:
+          path: .ndk/android-ndk-${{ env.NDK_CHANNEL }}
+          key: ndk-${{ env.NDK_CHANNEL }}
+
+      - name: Download NDK (if cache miss)
+        if: steps.cache-ndk.outputs.cache-hit != 'true'
+        run: |
+          set -euxo pipefail
+          mkdir -p .ndk
+          curl -L -o ndk.zip "https://dl.google.com/android/repository/android-ndk-${NDK_CHANNEL}-linux.zip"
+          unzip -q ndk.zip -d .ndk
+          rm ndk.zip
+
+      - name: Set NDK env + PATH
+        run: |
+          echo "NDK=${GITHUB_WORKSPACE}/.ndk/android-ndk-${NDK_CHANNEL}" >> $GITHUB_ENV
+          echo "${GITHUB_WORKSPACE}/.ndk/android-ndk-${NDK_CHANNEL}/toolchains/llvm/prebuilt/linux-x86_64/bin" >> $GITHUB_PATH
```

Comment:
Do you think it worth bothering with all this "use external cached NDK" instead of using the one available on `ubuntu-latest` under `ANDROID_NDK_ROOT`?

Since you explicitly specify the `ANDROID_ABI`/`ANDROID_API` - that shoud be enough for as long as `ANDROID_NDK_ROOT` is up-to-date enough (and right now it is [`27.3.13750724`](https://github.com/libusb/libusb-cmake/actions/runs/17261381779/job/48983433050)).

I had no issues using it this was for a while: https://github.com/libusb/libusb-cmake/blob/000e06eb241927db0f34e6867637d39b849ab989/.github/workflows/builds.yml#L219

Reply:
As of this moment: https://github.com/actions/runner-images/blob/57f0dc0e78cf05825b63f0a4b63932fa93b4ba3c/images/ubuntu/Ubuntu2404-Readme.md#android

Reply:
It seems to work @Youw  https://github.com/alexbirkett/libusb/actions/runs/17467661897